### PR TITLE
kubelet/userns: Add more unit tests

### PR DESCRIPTION
**Pull request by rata**
This covers all public methods and overall test coverage is above 80% again.

#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.: https://kep.k8s.io/127

```docs

```

### CODE REVIEW

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by rata** [comment]
This trivial PR is open for a month now. What can I do push this forward?

cc  @mrunalp @SergeyKanzhelev

File: pkg/kubelet/userns/userns_manager_test.go [hunk 1]
```diff
@@ -171,3 +183,179 @@ func TestUserNsManagerParseUserNsFile(t *testing.T) {
...
+	falseVal := false
+
+	cases := []struct {
+		name    string
+		pod     *v1.Pod
+		expMode runtimeapi.NamespaceMode
+		success bool
+	}{
+		{
+			name:    "no useer namespace",
```
Comment (mrunalp) [inline pkg/kubelet/userns/userns_manager_test.go:200]: typo: useer -> user

File: pkg/kubelet/userns/userns_manager_test.go [hunk 2]
```diff
@@ -171,3 +183,179 @@ func TestUserNsManagerParseUserNsFile(t *testing.T) {
...
+
+			for _, pod := range tc.podUnsetAfterCleanup {
+				ok := m.podAllocated(pod)
+				assert.False(t, ok, "pod %q should not be allocated", pod)
+			}
+		})
+	}
+}
+
+func TestAllocateOneMaxPods(t *testing.T) {
```
Comment (mrunalp) [inline pkg/kubelet/userns/userns_manager_test.go:327]: nit:  Maybe call it TestMaxPodsAllocation

Comment (mrunalp) [reply pkg/kubelet/userns/userns_manager_test.go:327]: or TestAllocateMaxPods like the next test.

Comment (rata) [reply pkg/kubelet/userns/userns_manager_test.go:327]: Fixed, thanks!

File: pkg/kubelet/userns/userns_manager_test.go (hunk 1 above)
Comment (rata) [reply pkg/kubelet/userns/userns_manager_test.go:200]: Fixed, thanks!

**Comment by rata** [comment]
@mrunalp fixed, PTAL

**Review by mrunalp — APPROVED** [review]
/lgtm

**Pull request merged**